Sources/Resource/AuthenticatedHTTPNetworkResource.swift

```diff
@@ -0,0 +1,24 @@
+import Foundation
+import struct Result.AnyError
+
+/// A type representing a resource that is fetched via HTTP using a specific type of endpoint requiring authentication.
+public protocol AuthenticatedHTTPNetworkResource: HTTPNetworkResource {
+
+    /// A type that represents a request authenticator.
+    associatedtype Authenticator: RequestAuthenticator where Authenticator.Request == Request
+
+    /// The resource's request authenticator.
+    var authenticator: Authenticator { get }
+}
+
+extension AuthenticatedHTTPNetworkResource {
+
+    @discardableResult
+    public func makeRequest(_ completion: @escaping MakeRequestCompletion) -> Cancelable {
+
+        return authenticator.authenticate(endpoint.request) { result in
+
+            return completion(result.mapError(AnyError.init))
```

Comment:
Unnecessary `return`

Reply:
Wrote it like this mostly for readability, but can definitely make it more compact 😉